### PR TITLE
pin coverage to 6.2

### LINKED ISSUE #5712: test_mp_pool_worker_no_daemon hangs after dependencies or CI env update
Overnight, with no changes to the distributed code, ``test_mp_pool_worker_no_daemon`` started hanging deterministically on Linux.

Last successful run: https://github.com/dask/distributed/runs/4944060002
First failed run: https://github.com/dask/distributed/runs/4949990263

Diff of mamba list on Python 3.8 doesn't show anything that catches the eye:
```
< coverage                  6.2              py38h497a2fe_0    conda-forge
> coverage                  6.3              py38h497a2fe_0    conda-forge

< distributed               2022.1.0+20.g682a7b1d           dev_0    <develop>
> distributed               2022.1.0+21.ge04c5627           dev_0    <develop>

< notebook                  6.4.7              pyha770c72_0    conda-forge
> notebook                  6.4.8              pyha770c72_0    conda-forge

< prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
> prometheus_client         0.13.0             pyhd8ed1ab_0    conda-forge
```
Note that the diff on Python 3.9 is a lot noiser due to a major change in pytorch which however is unrelated to the issue.

### PR DESCRIPTION
**Pull request by graingert**
Fixes https://github.com/dask/distributed/issues/5712

see https://github.com/nedbat/coveragepy/issues/1307
and https://github.com/nedbat/coveragepy/issues/1310
and https://github.com/nedbat/coveragepy/issues/1312

- [x] Closes #5712
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by crusaderky** [comment]
I think our issue should remain open until upstream is fixed

File: continuous_integration/environment-3.7.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ dependencies:
   - bokeh
   - click
   - cloudpickle
-  - coverage
+  - coverage<6.3
```
Comment (crusaderky) [inline continuous_integration/environment-3.7.yaml:13]:
```suggestion
  - coverage<6.3  # https://github.com/nedbat/coveragepy/issues/1310
```

File: continuous_integration/environment-3.8.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ dependencies:
   - bokeh
   - click
   - cloudpickle
-  - coverage
+  - coverage<6.3
```
Comment (crusaderky) [inline continuous_integration/environment-3.8.yaml:13]:
```suggestion
  - coverage<6.3  # https://github.com/nedbat/coveragepy/issues/1310
```

File: continuous_integration/environment-3.9.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ dependencies:
   - bokeh
   - click
   - cloudpickle
-  - coverage
+  - coverage<6.3
```
Comment (crusaderky) [inline continuous_integration/environment-3.9.yaml:14]:
```suggestion
  - coverage<6.3  # https://github.com/nedbat/coveragepy/issues/1310
```

**Review by crusaderky — CHANGES_REQUESTED** [review]

**Review by crusaderky — APPROVED** [review]

**Comment by crusaderky** [comment]
Thank you!

**Comment by jrbourbeau** [comment]
Great work @graingert!

**Pull request merged**